src/opera/pge/base/runconfig.py

```diff
@@ -252,6 +280,27 @@ def sas_schema_path(self) -> str:
             else resource_filename('opera', sas_schema_path)
         )
 
+    @property
+    def algorithm_parameters_schema_path(self) -> str:
+        """Returns the path to the algorithm parameter Schema file for DSWX-S1"""
+        algorithm_parameters_schema_path = self._pge_config['PrimaryExecutable']['AlgorithmParametersSchemaPath']
+        return (
```

Comment:
Since `AlgorithmParametersSchemaPath` has been added as an optional field, this method needs to account for the case where no value was provided at all. As is, a value of `None` would trigger the `resource_filename` call, which is not what we want. I think this method needs to check `algorithm_parameters_schema_path` is not `None`, and only then should it move on to the `return` logic.

If `algorithm_parameters_schema_path` is `None`, then that should be returned and its up to the caller to check if it got a result back.